packages/mantine/src/components/table/use-table.ts

```diff
@@ -213,21 +219,120 @@ export const useTable = <TData>(userOptions: UseTableOptions<TData> = {}): Table
     const options = defaultsDeep({}, userOptions, defaultOptions) as UseTableOptions<TData>;
     const initialState = defaultsDeep({}, options.initialState, defaultState) as TableState<TData>;
 
-    const [pagination, setPagination] = useState<TableState<TData>['pagination']>(initialState.pagination);
+    // synced with url
+    const [pagination, setPagination] = useUrlSyncedState<TableState<TData>['pagination']>({
```

Comment:
It looked a bit daunting at first with all those `useUrlSyncedState` but I like that each property is scoped to a single call.

Have you check to make sure it doesn't re-render 100x?

Reply:
Yeah the fact that each property of the state serializes to a different shape does not help here for the conciseness of the code. What I like with this implementation is that the logic for one state property is all in the same place and does not influence the other properties.

Reply:
It does not influence the number of rendering loops, because it does not set the state anymore often than a regular setState. The only time the url can influence the state is on initial render before calling `useState` so it does not create any additional rendering loops there either 🎉